JumpscaleCore/clients/ssh/SSHClientBase.py

```diff
@@ -45,6 +45,15 @@ def _init(self, **kwargs):
         self.executor = j.tools.executor.ssh_get(self)
         self._wireguard = None
         self._init3()
+        if (
+            j.sal.process.execute("ssh-add -l", die=False, showout=False)[2]
+            == "Could not open a connection to your authentication agent.\n"
+        ):
+            j.core.myenv.sshagent.start()
+            try:
```

Comment:
Instead of a try/except, I think it's better to do this:
```
if self.sshkey_name:
    self.sshkey_obj.load()
else:
    j.clients.sshkey.get().load()
```
but actually I am not sure if we should load the default key if no key is specified. 
@zaibon is it needed or should we just load the key if specified?

Reply:
This the ssh-add manually is wrong.  there is `j.clients.sshagent` module that has all the method to deal with ssh-agent. Both for listing loaded keys and listing new one.